### PR TITLE
Issue #86: fix starting of Che addon by default

### LINKED ISSUE #86: Che Minishift addon out of the box is broken
Che Minishift addon doesn't deploy Che properly. 
I believe it is related to the fact that upstream Che switched Che 5 and Che 6 branches and now Che 6 is in master whereas Che Minishift addon expects it to be Che 5. 
Here is the image declaration that is built by CI from master right now https://github.com/minishift/minishift-addons/blob/master/add-ons/che/templates/che-single-user.yml#L27.
Also, as @l0rd suggested we need a CI to fix a situation where Che addon is broken and devs don't know about that.

### PR DESCRIPTION
**Pull request by garagatyi**
Fixed start of Che in Che minishift addon. Cleaned up and aligned docs and code according to the current state of deployment of Che. Now Che 6 starts by default.

Checked start of Che 6 and Che 5.

Fixes #86

### CODE REVIEW

**Comment by garagatyi** [comment]
@l0rd @gbraad Please, take a look. This PR follows our discussion from PR #87

**Comment by budhrg** [comment]
I am able to deploy che addon. However, not able to create any stack. Use Java, Java+SQL
screenshot from 2018-02-15 16-01-05
screenshot from 2018-02-15 16-02-25
screenshot from 2018-02-15 16-14-10

**Comment by garagatyi** [comment]
About MySQL - you have probably used compose recipe which is not yet supported on OpenShift in Che 6.
As to the other error, I'll investigate it. Can you provide command line you have used to apply Che addon resources?

**Comment by garagatyi** [comment]
I can reproduce the second error. I'll work on it. Thank you @budhrg !

**Comment by garagatyi** [comment]
TLDR: there is a bug, I added a workaround in the PR. Please, recheck.
@budhrg it is a hilarious story! 
Yesterday it worked and today it is not. And there is a reason for that.
There is a tiny bug in the usage of properties in Che. I created [PR](https://github.com/eclipse/che/pull/8795) to fix this.
I use latest Che tag in this PR which uses latest Che release, not the latest code in the master branch.
And since release with the bug happened last night I pulled code without bug yesterday and you pulled new code with the bug today!
And since Che addon depends on the latest release, not the latest code from Che master we will not be able to fix that properly ASAP - we will have to wait for a bugfix release after my PR is merged.
To workaround that I added kubernetes infra related property which fixes start of workspaces right now. We can remove this property after the Che bugfix release.

**Comment by budhrg** [comment]
> Yesterday it worked and today it is not. And there is a reason for that.
There is a tiny bug in the usage of properties in Che. I created PR to fix this.
I use latest Che tag in this PR which uses latest Che release, not the latest code in the master branch.
And since release with the bug happened last night I pulled code without bug yesterday and you pulled new code with the bug today!

Indeed a hilarious story :laughing: 

> To workaround that I added kubernetes infra related property which fixes start of workspaces right now. 

Will test tomorrow.

> We can remove this property after the Che bugfix release.

When we can expected that?

**Comment by garagatyi** [comment]
I can do that as soon as Che release with my PR done. Would take a couple of days I suppose. But PR works right now because of workaround I added. So it is safe to merge PR and clean up the workaround after Che release.

**Comment by budhrg** [comment]
> So it is safe to merge PR and clean up the workaround after Che release.

If it is not taking greater than a week, then we should would. WDYT @LalatenduMohanty ?

**Comment by garagatyi** [comment]
Release manager of Che says that they don't have any particular plans to release new bugfix release yet. 
Any updates from you on this PR? Should I try to promote release or we can merge this PR and wait as long as it is needed for that release?

**Comment by LalatenduMohanty** [comment]
> If it is not taking greater than a week, then we should would. WDYT @LalatenduMohanty ?

We will merge this as the add-on is working with the workaround. Do not want to have a broken add-on ;-)

**Comment by l0rd** [comment]
@LalatenduMohanty I think there is a misunderstanding. The add-on is currently broken and this PR fixes it. There was a bug in latest Che that broke this PR but a subsequent @garagatyi commit has fixed it. So the sooner you merge this PR the better :bowtie:

**Comment by budhrg** [comment]
@garagatyi 
I tested with Java stack and able start workspace and perform build and run sample app.

Didn't tested Java+MySQL stack as you mentioned some known issue here https://github.com/minishift/minishift-addons/pull/88#issuecomment-365904303.

LGTM :+1:
